api/src/pcapi/serialization/decorator.py

```diff
@@ -132,11 +140,20 @@ def sync_validate(*args: Any, **kwargs: Any) -> Response:
             try:
                 body_params = request.get_json()
             except BadRequest as error:
-                logger.info(
-                    "Error when decoding request body: %s",
-                    error,
-                    extra={"contentTypeHeader": request.headers.get("Content-Type"), "path": request.path},
-                )
+                try:
+                    content_length = request.headers.get("Content-Length", 1000, type=int)
+                except ValueError:
+                    extra = {"contentTypeHeader": request.headers.get("Content-Type"), "path": request.path}
+                else:
+                    extra = {"contentTypeHeader": request.headers.get("Content-Type"), "path": request.path}
+                    if content_length is not None and content_length < 1000:
+                        extra = {
+                            "contentTypeHeader": request.headers.get("Content-Type"),
+                            "path": request.path,
+                            "data": request.get_data(as_text=True),
+                        }
```

Comment:
(détail / style) C'est un peu moins répétitif si on définit `extra` dès le début du bloc (et qu'on l'augmente si on peut extraire plus d'informations). Au passage, je suggère de logger le "content-length". Ainsi, si on s'aperçoit que peu de requêtes sont loggées et que beaucoup sont entre 1000 et 5000 octets, on pourrait augmenter le seuil à 5000 caractères pour plus logger et mieux comprendre les requêtes invalides.

```suggestion
                extra = {
                    "contentTypeHeader": request.headers.get("Content-Type"),
                    "path": request.path,
                    "lengthHeader": request.headers.get("Content-Length", "missing header")[:20],
                    }
                # Extract payload if it's not too large.
                try:
                    content_length = request.headers.get("Content-Length", 1000, type=int)
                except ValueError:
                    pass
                else:
                    if content_length is not None and content_length < 1000:
                        extra["data"] = request.get_data(as_text=True)
```

Reply:
(discussion) J'ai un doute sur `request.get_data(as_text=True)`. Si le client fournit des données qui ne sont pas encodées avec l'encodage indiqué dans l'en-tête HTTP qui va bien, je pense que cette appel échouera. C'est peu probable... mais comme on parle de clients qui envoient du JSON invalides, autant prendre des précautions. :) Je pense qu'on peut se contenter de logger des bytes : 

```diff
- extra["data"] = request.get_data(as_text=True)
+ extra["data"] = request.get_data()
```

... ça [échouera si le client n'envoie pas de l'UTF-8](https://github.com/pass-culture/pass-culture-main/blob/833467e2e3aaf3b22b55e7fc0ae557b5b0fe1864/api/src/pcapi/core/logging.py#L148-L149), mais au moins ça échouera dans notre code de logging et pas ici.